boltdesignsystem/bolt
# Feat/bds 1937 add compatibility mode

## Jira

http://vjira2:8080/browse/BDS-1937

## Summary

Adds a compatibility flag that determines whether a webpack build is necessary when esModules is enabled.

## Details

If both `prod` is false and `esmodules` is true, by default local builds will only get the "modern" webpack bundle (as defined in create-webpack-config.js).  This dramatically improves build time while allowing local development for most situations. 

If, however, you want to test or debug the legacy build while working locally (the most obvious use case is if you want to troubleshoot IE 11), this PR allows you to do so by running `yarn run start:compat` (under the hood, `yarn bolt start --compat`).

## How to test

On this feature branch:
- Run `yarn run start:compat`.  Confirm that you see two webpack build progress indicators in the browser while starting (and that the docs site works in IE 11)
- Run `yarn run start`.  Confirm that only one webpack build runs (and that the docs site does not works in IE 11)
